python/pyarrow/cffi.py

```diff
@@ -64,6 +64,16 @@
       // Opaque producer-specific data
       void* private_data;
     };
+
+    typedef int32_t ArrowDeviceType;
```

Comment:
should we expose the constants in pyarrow somehow?

Reply:
If we don't use them ourselves, I don't know if that is needed (although it might still be useful for other _users_ of `pyarrow.cffi`?)